### DISCUSSION TITLE
Shrink neighboring grid item instead of pushing it down

### DISCUSSION THREAD

**Discussion by i0000010** [react-grid-layout/react-grid-layout#1674] [Q&A]
Say I have two grid items placed side by side and I want keep it that way. If I resize the width of the item on the left side so that it surpasses the x value of the item on the right side, then the item on the right gets pushed down. Is there a way to modify this behavior so that the width of the item on the right size shrinks instead of the item getting pushed down?

The idea is that I want to create an adjustable width sidebar such that the width of the main content shrinks as the side bar expands and vice versa.

**Comment by STRML** [comment]
There's no support for automatically resizing other items on collisions, only moving them. You can change the behavior by using horizontal compaction.
